agent/lib/kontena/observer.rb

```diff
@@ -68,6 +78,10 @@ def update_observe(observe, observable, value)
     # Yield happens once each Observable is ready, and whenever they are updated.
     # Yields to block happens from different async tasks.
     #
+    # The block must be idempotent: it is not guaranteed to receive every Observable update.
+    # It is only guaranteed to receive later Observable updates in the correct order,
+    # and it may receive some Observable updates multiple times.
```

Comment:
This naturally happens if the Observer crashes, restarts and re-observes, but it can also happen during the initial `observe` cycle, if the initial `add_observer` returns nil, but then the observable updates with some value before the initial `async.observed` happens

```
D, [2017-03-31T17:05:09.484644 #13147] DEBUG -- : observer: Observer<> <- nil...
D, [2017-03-31T17:05:09.485523 #13147] DEBUG -- : update: 1
D, [2017-03-31T17:05:09.485864 #13147] DEBUG -- : observe Observable<#<Class:0x00000002557e98>> -> 
D, [2017-03-31T17:05:09.486180 #13147] DEBUG -- : notify: Observer<> <- 1...
D, [2017-03-31T17:05:09.499476 #13147] DEBUG -- : observe Observable<#<Class:0x00000002557e98>> -> 1
D, [2017-03-31T17:05:09.513581 #13147] DEBUG -- : observed  -> 1
D, [2017-03-31T17:05:09.520398 #13147] DEBUG -- : observed 1 -> 1
```